File: memoria_fisica.cpp
#include "memoria_fisica.hpp"

MemoriaFisica::MemoriaFisica(int pTamanho) {
	this->memoria_fisica = (int *) malloc(sizeof(int) * pTamanho);
}

int *MemoriaFisica::getMemoriaFisica() {
	return this->memoria_fisica;
}

int MemoriaFisica::getValorMemoriaFisica(int pIndice) {
	return this->memoria_fisica[pIndice];
}

void MemoriaFisica::setValorMemoriaFisica(int pValor, int pIndice) {
	this->memoria_fisica[pIndice] = pValor;
}

File: src/memoria/MemoriaFisica.java
package memoria;

public class MemoriaFisica {	
	private Integer[] memoriaFisica;
	private int tamanho = 0;
	
	public MemoriaFisica(int pTamanho) {
		this.tamanho = pTamanho;
		memoriaFisica = new Integer[pTamanho];
	}
	
	public Integer getValor(int pIndice) {
		return memoriaFisica[pIndice];
	}
	
	public void setValor(int pIndice, Integer pValor) {
		this.memoriaFisica[pIndice] = pValor;
	}
	
	public int getTamanho() {
		return this.tamanho;
	}
	
	public Integer[] getMemoria() {
		return this.memoriaFisica;
	}
	
	public Integer getIndiceLivre(Integer[] memFisica) {
		for(int i = 0; i < memFisica.length; i++){
			if(memFisica[i] == null){
				return i;
			}
		}
		return null;
	}
}